crypto/rand/drbg_hmac.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2011-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+#include "internal/thread_once.h"
+#include "rand_lcl.h"
+
+/* Called twice by SP800-90Ar1 10.1.2.2 HMAC_DRBG_Update_Process.
+ * inbyte is 0x00 on the first call and 0x01 on the second call.
+ */
+static int do_hmac(RAND_DRBG_HMAC *hmac, unsigned char inbyte,
+                   const unsigned char *in1, size_t in1len,
+                   const unsigned char *in2, size_t in2len,
+                   const unsigned char *in3, size_t in3len)
+{
+    HMAC_CTX *ctx = hmac->ctx;
+           /* K = HMAC(K, V || inbyte || provived_data */
+    return HMAC_Init_ex(ctx, hmac->K, hmac->blocklen, hmac->md, NULL)
```

Comment:
Why not EVP?

Reply:
I had a look at what would be required to do this..
It might be best to leave this alone, until we revisit the mac EVP interface. I think its currently a bit of a mess hijacking the MD_CTX and then shoving it into SignDigest. The hmac interface is much simpler and nicer to interface to currently (which is probably why it is used in multiple other places also - such as hkdf.c)